### PR TITLE
Add dartdoc to VirtualViewport widgets

### PR DESCRIPTION
**Pull request by abarth**

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter/lib/src/widgets/virtual_viewport.dart [hunk 1]
```diff
@@ -11,7 +11,9 @@ import 'package:flutter/rendering.dart';
 
 typedef void ExtentsChangedCallback(double contentExtent, double containerExtent);
 
+/// An abstract widget whose children are not all materialized.
 abstract class VirtualViewport extends RenderObjectWidget {
+  /// The offset at which the
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/virtual_viewport.dart]: at which the what!

File: packages/flutter/lib/src/widgets/virtual_viewport.dart [hunk 2]
```diff
@@ -162,7 +182,12 @@ abstract class VirtualViewportElement<T extends VirtualViewport> extends RenderO
   }
 }
 
-abstract class VirtualViewportIterableMixin extends VirtualViewport {
+/// A VirtualViewport that represents its children using [Iterable<Widget>].
+///
+/// The iterator is advanced just far enough to obtain widgets for the children
+/// that need to be materialized.
+abstract class VirtualViewportIterable extends VirtualViewport {
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/virtual_viewport.dart]: Should this be "IterableVirtualViewport" or "IteratingVirtualViewport"? "VirtualViewportIterable" sounds like it's an Iterable, which it isn't.

File: packages/flutter/lib/src/widgets/virtual_viewport.dart [hunk 3]
```diff
@@ -205,10 +230,21 @@ class _IterableWidgetProvider extends _WidgetProvider {
   Widget getChild(int i) => _widgets[(i % _length).abs()];
 }
 
+/// Returns the sublist of widgets in the given range.
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/virtual_viewport.dart]: "Signature of a callback that returns..."

File: packages/flutter/lib/src/widgets/virtual_viewport.dart [hunk 4]
```diff
@@ -11,7 +11,9 @@ import 'package:flutter/rendering.dart';
 
 typedef void ExtentsChangedCallback(double contentExtent, double containerExtent);
 
+/// An abstract widget whose children are not all materialized.
 abstract class VirtualViewport extends RenderObjectWidget {
+  /// The offset at which the viewport should start painting children.
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/virtual_viewport.dart]: offset from where? This should probably at least reference the anchor property and its type (which has the details).

**Comment by Hixie** [comment]
lgtm

**Pull request merged**